presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -258,6 +260,8 @@ public PlanNode visitProject(ProjectNode node, RewriteContext<Expression> contex
 
             List<Expression> inlinedDeterministicConjuncts = inlineConjuncts.get(true).stream()
                     .map(entry -> inlineSymbols(node.getAssignments().getMap(), entry))
+                    .map(conjunct -> canonicalizeExpression(conjunct, typeAnalyzer.getTypes(session, types, conjunct), metadata))
```

Comment:
add comment why this is needed?
BTW: why canonicalization is needed here?

Reply:
It's because UnwrapCasts only works with expressions of the form `CAST(x AS T) = constant`. `canonicalizeExpression` converts `constant = CAST(x AS T)`